hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerStateManagerImpl.java

```diff
@@ -275,12 +278,24 @@ public void addContainer(final ContainerInfoProto containerInfo)
 
     if (!containers.contains(containerID)) {
       ExecutionUtil.create(() -> {
-        containerStore.put(containerID, container);
+        if (transactionBuffer != null) {
+          containerStore.putWithBatch(
+              transactionBuffer.getCurrentBatchOperation(),
+              containerID, container);
+        } else {
+          containerStore.put(containerID, container);
```

Comment:
This is leave for MockHAManager code path in which the buffer is NULL :) 

If we choose to update MockHAManager to have a sort of MockBuffer implementation, then we can move this `else branch` because at that time I think all existing code will access this buffer.

I can create a JIRA to track that add buffer to MockHAManager and remove this else :-)

Reply:
https://issues.apache.org/jira/browse/HDDS-4634

Create JIRA to add buffer in MockHAManager thus we can remove this `null` check.